systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/KafkaAdminClient.java

```diff
@@ -28,101 +29,64 @@
+        replicationFactor = (builder.getReplicationFactor() <= 0) ? 1 : builder.getReplicationFactor();
+        partitions = (builder.getPartitions() <= 0) ? 1 : builder.getPartitions();
+        topicCount = (builder.getTopicCount() <= 0) ? 1 : builder.getTopicCount();
+        topicOffset = builder.getTopicOffset();
+        adminName = builder.getAdminName();
+        topicOperation = builder.getTopicOperation();
     }
 
-    public String getTopicName() {
-        return topicName;
+    @Override
+    protected KafkaAdminClientsBuilder newBuilder() {
+        return new KafkaAdminClientsBuilder();
     }
 
-    public int getTopicCount() {
-        return topicCount;
-    }
+    protected KafkaAdminClientsBuilder updateBuilder(KafkaAdminClientsBuilder builder) {
+        super.updateBuilder(builder);
 
-    public int getTopicOffset() {
-        return topicOffset;
-    }
-
-    public String getAdditionalConfig() {
-        return additionalConfig;
-    }
-
-    public String getNamespaceName() {
-        return namespaceName;
-    }
-
-    public int getPartitions() {
-        return partitions;
+        return builder
+            .withTopicOperation(topicOperation)
+            .withTopicCount(topicCount)
+            .withTopicOffset(topicOffset)
+            .withPartitions(partitions)
+            .withReplicationFactor(replicationFactor)
+            .withKafkaAdminClientName(adminName);
     }
 
-    public int getReplicationFactor() {
-        return replicationFactor;
+    @Override
+    public KafkaAdminClientsBuilder toBuilder() {
+        return updateBuilder(newBuilder());
     }
 
-    public String getTopicOperation() {
-        return topicOperation;
-    }
+    private void checkParametersForBuilder(KafkaAdminClientsBuilder builder) {
+        if (builder.getTopicOperation() == null || builder.getTopicOperation().isEmpty()) {
+            throw new InvalidParameterException("TopicOperation must be set.");
+        }
 
-    public KafkaAdminClient.Builder newBuilder() {
-        return new KafkaAdminClient.Builder();
-    }
+        if (builder.getBootstrapAddress() == null || builder.getBootstrapAddress().isEmpty()) {
+            throw new InvalidParameterException("Bootstrap server is not set.");
+        }
 
-    public KafkaAdminClient.Builder updateBuilder(KafkaAdminClient.Builder builder) {
-        return builder
-            .withAdditionalConfig(getAdditionalConfig())
-            .withBootstrapAddress(getBootstrapAddress())
-            .withTopicOperation(getTopicOperation())
-            .withTopicCount(getTopicCount())
-            .withTopicOffset(getTopicOffset())
-            .withPartitions(getPartitions())
-            .withReplicationFactor(getReplicationFactor())
-            .withKafkaAdminClientName(getAdminName())
-            .withTopicName(getTopicName())
-            .withNamespaceName(getNamespaceName());
+        if ((builder.getTopicName() == null || builder.getTopicName().isEmpty())
```

Comment:
Here you just need `(builder.getTopicOperation().equals("help") || builder.getTopicOperation().equals("list"))`

Reply:
thanks! good catch :)

Reply:
Discussed offline, this has to be like it is.